eland/ml/pytorch/pytorch_model.py

```diff
@@ -0,0 +1,154 @@
+        with open(path, "r") as infile:
+            return json.load(infile)
+
+    def _upload_config(self, path: str) -> bool:
+        config = PyTorchModel._load_json(path)
+        response = self._client.ml.put_trained_model(
+            model_id=self.model_id, body=config
+        )
+        return response is not None
+
+    def _upload_vocab(self, path: str) -> bool:
+        vocab = PyTorchModel._load_json(path)
+        return self._client.transport.perform_request(
+            method="PUT",
+            url=f"/_ml/trained_models/{self.model_id}/vocabulary",
+            body=vocab,
+        )
+
+    def _upload_model(
+        self, model_path: str, chunk_size: int = DEFAULT_CHUNK_SIZE
+    ) -> bool:
+        file_stats = os.stat(model_path)
+        total_parts = math.ceil(file_stats.st_size / chunk_size)
+
+        def model_file_chunk_generator():
+            with open(model_path, "rb") as f:
+                while True:
+                    data = f.read(chunk_size)
+                    if not data:
+                        break
+                    yield base64.b64encode(data).decode()
+
+        for i, data in tqdm(
+            enumerate(model_file_chunk_generator(), start=0), total=total_parts
+        ):
+            body = {
+                "total_definition_length": file_stats.st_size,
+                "total_parts": total_parts,
+                "definition": data,
+            }
+            self._client.transport.perform_request(
+                method="PUT",
+                url=f"/_ml/trained_models/{self.model_id}/definition/{i}",
+                body=body,
+            )
+
+        return True
+
+    def upload(
+        self,
+        model_path: str,
+        config_path: str,
+        vocab_path: str,
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+    ) -> bool:
+        # TODO: Implement some pre-flight checks on config, vocab, and model
+        return (
+            self._upload_config(config_path)
+            and self._upload_vocab(vocab_path)
+            and self._upload_model(model_path, chunk_size)
+        )
+
+    def start(self, timeout: str = DEFAULT_TIMEOUT) -> bool:
+        return self._client.transport.perform_request(
+            method="POST",
+            url=f"/_ml/trained_models/{self.model_id}/deployment/_start",
+            params={"timeout": timeout, "wait_for": "started"},
+        )
+
+    def stop(self) -> bool:
+        try:
+            return self._client.transport.perform_request(
+                method="POST",
+                url=f"/_ml/trained_models/{self.model_id}/deployment/_stop",
+                params={"ignore": 404},
+            )
+        except elasticsearch.NotFoundError:
+            pass
+
+    def delete(self) -> Dict[str, Any]:
+        try:
+            return self._client.ml.delete_trained_model(self.model_id, ignore=(404,))
+        except elasticsearch.NotFoundError:
+            pass
+
+    @staticmethod
+    def list(
```

Comment:
Since this isn't doing a ton beyond what the client is already doing, thoughts on removing this method?

Reply:
I wanted to have start, stop, delete all in the same place since right now the client doesn't have start and stop on trained model. So having them all here is just consistent until they are all available in the client.

Reply:
I can remove `list` since I only really use it in a "cleanup" script for QA.